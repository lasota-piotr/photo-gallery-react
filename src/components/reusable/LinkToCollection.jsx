import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import slugify from '../../helpers/slugify';

const LinkToCollection = ({ children, id, title, state }) => (
  <Link
    to={{
      pathname: `/collections/${id}/${slugify(title).slice(0, 25)}`,
      state,
    }}
  >
    {children}
  </Link>
);

LinkToCollection.propTypes = {
  children: PropTypes.node.isRequired,
  id: PropTypes.oneOfType([PropTypes.string, PropTypes.number]).isRequired,
  title: PropTypes.string.isRequired,
};

export default LinkToCollection;
