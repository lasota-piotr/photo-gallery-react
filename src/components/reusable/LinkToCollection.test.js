import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import LinkToCollection from './LinkToCollection';

test("Generates anchor with url to collection's page", () => {
  const component = renderer.create(
    <MemoryRouter>
      <LinkToCollection id="123456" title="test title - @">Test link</LinkToCollection>
    </MemoryRouter>
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
