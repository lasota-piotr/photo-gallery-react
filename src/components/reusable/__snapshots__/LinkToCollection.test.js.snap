// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Generates anchor with url to collection's page 1`] = `
<a
  href="/collections/123456/test-title"
  onClick={[Function]}
>
  Test link
</a>
`;
